snuba/migrations/snuba_migrations/metrics/0019_aggregate_tables_add_ttl.py

```diff
@@ -0,0 +1,45 @@
+from typing import Sequence
+
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    """
+    Adds TTLs to all the aggergate tables based on retention_days columns
+    """
+
+    blocking = False
+    table_names = [
+        "metrics_distributions_local",
+        "metrics_sets_local",
+        "metrics_counters_local",
+    ]
+
+    def forwards_local(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.RunSql(
+                storage_set=StorageSetKey.METRICS,
+                statement=(
+                    f"ALTER TABLE {table_name} "
+                    "MODIFY TTL "
+                    "if(isNull(retention_days), timestamp + toIntervalDay(90), timestamp + toIntervalDay(retention_days))"
+                ),
+            )
+            for table_name in self.table_names
+        ]
+
+    def backwards_local(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.RunSql(
+                storage_set=StorageSetKey.METRICS,
+                statement=(f"ALTER TABLE {table_name} REMOVE TTL"),
```

Comment:
Is this the change that requires 21.8 ?
SO what is the plan ?

Reply:
It's @lynnagara's preference to just not have a backwards migration here rather than support separate minimum versions